### ISSUE TITLE
Unable to reproduce the experimental results from the paper

### ISSUE THREAD

**Issue by typist2001** [minwoo0611/HeLiOS#14]
We are unable to reproduce the experimental results reported in the paper when retraining the model from scratch.

We have tried both the default configuration and the following training setup:
[DEFAULT]
dataset_folder = data

[TRAIN]
model_name = HeLiOS_default_trainning
num_workers = 4
batch_size = 2048
batch_split_size = 200
val_batch_size = 256

lr = 5e-4
epochs = 150
scheduler_milestones = 40, 70
aug_mode = 1
weight_decay = 1e-4
train_file = training.pickle
val_file = validation.pickle
eval_db_file = Roundabout_evaluation_db.pickle
eval_query_file = Roundabout_evaluation_query.pickle

[LOSS]
tau1 = 0.01
positives_per_query = 4
p_sp_distance = 0.02
sp_n_distance = 0.19
truncated_weight = 0.8
guided_p_sp_weight = 0.1
guided_sp_n_weight = 0.1

However, we observe that both configurations start to overfit after around 20 epochs.

Although we are able to reproduce the results reported in the paper when using the provided pretrained weights, retraining the model from scratch leads to a performance gap of approximately 10% compared to the reported results.

Could you please provide guidance on how to resolve this issue?

Thank you very much for your help.

**Comment by minwoo0611** [comment]
Hi @typist2001 

Thank you for sharing your detailed experimental setup.

Regarding the reproduction issues, we have confirmed that the results are reproducible under specific configurations. You can find the detailed logs and reproduced results in Issue #12. We have verified the reproduction with a batch_split_size of 32.

As you observed, overfitting is a known issue if the model is trained until the final epoch defined in the default configuration. To address the performance gap and mitigate overfitting, I recommend the following:

1. Validation and Evaluation Strategy
The current Validation AP tends to saturate at high values, making it difficult to monitor the onset of overfitting accurately. I suggest using the evaluation code to check the actual performance at regular iteration intervals. Monitoring the evaluation metrics more frequently will help identify the optimal checkpoint before overfitting occurs.

2. Learning Rate Adjustment
The default learning rate is not an optimized value. I recommend using a lower learning rate, to ensure more stable convergence.

3. Training Schedule
Since the model begins to overfit early in your setup, I do not recommend training until the end of the predefined epochs. Please consider using an Early Stopping strategy based on the evaluation results mentioned above. If you require immediate deployment, using the provided pre-trained weights is the most reliable option at this stage.

Having a nice Christmas!

Best regards,

Minwoo Jung

**Comment by minwoo0611** [comment]
Hi @typist2001 

Thank you for sharing your detailed experimental setup.

Regarding the reproduction issues, we have confirmed that the results are reproducible under specific configurations. You can find the detailed logs and reproduced results in Issue #12. We have verified the reproduction with a batch_split_size of 32.

As you observed, overfitting is a known issue if the model is trained until the final epoch defined in the default configuration. To address the performance gap and mitigate overfitting, I recommend the following:

1. Validation and Evaluation Strategy
The current Validation AP tends to saturate at high values, making it difficult to monitor the onset of overfitting accurately. I suggest using the evaluation code to check the actual performance at regular iteration intervals. Monitoring the evaluation metrics more frequently will help identify the optimal checkpoint before overfitting occurs.

2. Learning Rate Adjustment
The default learning rate is not an optimized value. I recommend using a lower learning rate, to ensure more stable convergence.

3. Training Schedule
Since the model begins to overfit early in your setup, I do not recommend training until the end of the predefined epochs. Please consider using an Early Stopping strategy based on the evaluation results mentioned above. If you require immediate deployment, using the provided pre-trained weights is the most reliable option at this stage.

Having a nice Christmas!

Best regards,

Minwoo Jung
